### ISSUE TITLE
Publish Command

### ISSUE THREAD

**Issue by BennyHallett** [BennyHallett/obelisk#6]
Need a command to publish the built site via various methods.

Task should do a `build` first to ensure it's up to date, but provide an option to not do a build.

```
$ mix obelisk publish
$ mix obelisk publish --no-build
```

Methods could include:
- SSH/SCP
- RSync
- Github Pages

**Comment by optikfluffel** [comment]
I know it's a little old-school but what about good old FTP? Oh and maybe S3 would be cool too? ^^

**Comment by BennyHallett** [comment]
Yeah that'd be cool. I think it should be fairly extensible, so that we can easily add new providers.
